lib/recipe/bloc/recipe_bloc.dart

```diff
@@ -150,13 +190,172 @@ class RecipeBloc extends Bloc<RecipeEvent, RecipeState> {
+        List.from(probes[changedProbeIndex].probeOptions);
+    int changedProbeOptionIndex = event.probeOptionIndex;
+
+    String probeOptionName = event.probeOptionName;
+    Map<String, String?> probeOption = {
+      'option': probeOptionName,
+      'id': probeOptions[changedProbeOptionIndex]['id']
+    };
+    probeOptions.removeAt(changedProbeOptionIndex);
+    probeOptions.insert(changedProbeOptionIndex, probeOption);
+
+    Probe probe =
+        probes[changedProbeIndex].copyWith(probeOptions: probeOptions);
+
+    probes.removeAt(changedProbeIndex);
+    probes.insert(changedProbeIndex, probe);
+
+    Recipe recipe = recipes[changedRecipeIndex].copyWith(probes: probes);
+
+    recipes.removeAt(changedRecipeIndex);
+    recipes.insert(changedRecipeIndex, recipe);
+
+    emit(state.copyWith(recipes: recipes));
+  }
+
+  void _onProbeOptionDeleted(
+      ProbeOptionDeleted event, Emitter<RecipeState> emit) {
+    List<Recipe> recipes = List.from(state.recipes);
+    int changedRecipeIndex = recipes.indexOf(event.recipe);
+
+    List<Probe> probes = List.from(recipes[changedRecipeIndex].probes);
+    int changedProbeIndex = event.probeIndex;
+    List<Map<String, dynamic>> probeOptions =
+        List.from(probes[changedProbeIndex].probeOptions);
+
+    probeOptions.removeAt(event.probeOptionIndex);
+
+    Probe probe =
+        probes[changedProbeIndex].copyWith(probeOptions: probeOptions);
+
+    probes.removeAt(changedProbeIndex);
+    probes.insert(changedProbeIndex, probe);
+
+    bool probesStandard = true;
+    for (Probe probe in probes) {
+      if (probe.standardAnswer() == false) {
+        probesStandard = false;
+        break;
+      }
+    }
+
+    Recipe recipe = recipes[changedRecipeIndex]
+        .copyWith(probes: probes, probesStandard: probesStandard);
+
+    recipes.removeAt(changedRecipeIndex);
+    recipes.insert(changedRecipeIndex, recipe);
+
+    emit(state.copyWith(recipes: recipes));
+  }
+
+  void _onProbeOptionSelected(
+      ProbeOptionSelected event, Emitter<RecipeState> emit) {
+    List<Recipe> recipes = List.from(state.recipes);
+    int changedRecipeIndex = recipes.indexOf(event.recipe);
+
+    List<Probe> probes = List.from(recipes[changedRecipeIndex].probes);
+    int changedProbeIndex = event.probeIndex;
+
+    Probe probe = probes[changedProbeIndex].copyWith(answer: event.answer);
+
+    probes.removeAt(changedProbeIndex);
+    probes.insert(changedProbeIndex, probe);
+
+    bool probesStandard = true;
+    for (Probe probe in probes) {
+      if (probe.standardAnswer() == false) {
+        probesStandard = false;
+        break;
+      }
+    }
+
+    Recipe recipe = recipes[changedRecipeIndex]
+        .copyWith(probes: probes, probesStandard: probesStandard);
+
+    recipes.removeAt(changedRecipeIndex);
+    recipes.insert(changedRecipeIndex, recipe);
+
+    emit(state.copyWith(recipes: recipes));
+  }
+
+  void _onProbeCleared(ProbeCleared event, Emitter<RecipeState> emit) {
```

Comment:
You almost got it right - the idea is that when a recipe is selected from the recipes screen, previous probe interaction (answers/checkboxes) are wiped. This is because a standard recipe might be used more than once for different food items/ collections and hence the recipe itself should not persist probe interaction data (it is instead stored with the food item)